.github/workflows/docker-push.yml

```
@@ -3,35 +3,62 @@ on: create
 
 jobs:
   release:
+    if: github.event.ref_type == 'tag'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.4.0
-        if: github.event.ref_type == 'tag'
 
       - uses: actions/setup-go@v2.1.5
-        if: github.event.ref_type == 'tag'
+
         id: go
         with:
           go-version: 1.17
 
       - name: go-build
-        if: github.event.ref_type == 'tag'
+
         run: go build -o fake-gcs-server
         env:
           CGO_ENABLED: 0
 
       - name: sanity-check
-        if: github.event.ref_type == 'tag'
         uses: docker://alpine
         with:
           entrypoint: sh
           args: -c "./fake-gcs-server -h"
 
-      - name: docker-publish
-        if: github.event.ref_type == 'tag'
-        uses: fsouza/docker-publish-action@v1.0.1
+      - name: docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ secrets.DOCKER_USERNAME }}/fake-gcs-server
```

```suggestion
          images: fsouza/fake-gcs-server
```

DOCKER_USERNAME isn't actually fsouza 🙈  I use a bot account with limited privileges (I know it can push, but... 😬), just in case it leaks heh

---

:)